### ISSUE TITLE
The release of 1.10.0 breaks the internet

### ISSUE THREAD

**Issue by jandppw** [auth0/node-jwks-rsa#179]
**Please do not report security vulnerabilities here**. The [Responsible Disclosure Program](https://auth0.com/whitehat) details the procedure for disclosing security issues.

**Thank you in advance for helping us to improve this library!** Please read through the template below and answer all relevant questions. Your additional work here is greatly appreciated and will help us respond as quickly as possible. For general support or usage questions, use the [Auth0 Community](https://community.auth0.com/) or [Auth0 Support](https://support.auth0.com/). Finally, to avoid duplicates, please search existing Issues before submitting one here.

By submitting an Issue to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Since a number of hours our builds fail with 

```
Error: Cannot find module 'https-proxy-agent'
Require stack:
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/node_modules/jwks-rsa/lib/wrappers/request.js
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/node_modules/jwks-rsa/lib/JwksClient.js
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/node_modules/jwks-rsa/lib/index.js
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/lib/server/authorization/publicSigningKeyGetter.js
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/test/_getTrueToken.js
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/test/unit/getTrueToken.test.js
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/node_modules/mocha/lib/esm-utils.js
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/node_modules/mocha/lib/mocha.js
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/node_modules/mocha/lib/cli/one-and-dones.js
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/node_modules/mocha/lib/cli/options.js
- /Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/node_modules/mocha/bin/mocha
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/node_modules/jwks-rsa/lib/wrappers/request.js:52:24)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
    at Module.replacementCompile (/Users/jand/Scratchpad/pink-ribbon-belgium/pink-march-service/node_modules/append-transform/index.js:60:13)
    at Module._extensions..js (internal/modules/cjs/loader.js:1158
...
```

This coincides with this release.

Our dependencies are:

```
  "dependencies": {
...
    "jwks-rsa": "^1.8.0",
...
  },
```

Once I did a `rm -Rf node_modules`, I now have a `jwks-rsa` whose `package.json` says

```
...
  "_from": "jwks-rsa@^1.8.0",
  "_id": "jwks-rsa@1.10.0",
...
```

and where `https-proxy-agent` is in the `devDependencies`, not in the dependencies.

### Reproduction

/

### Environment

* macOS
* Bitbucket Pipelines

**Comment by jandppw** [comment]
Yep.

The workaround is to do

```
npm install jwks-rsa@1.8.0 --save-exact
```

The culprit is indeed that you added `require('https-proxy-agent')` to `lib/wrappers/request.js`, while `https-proxy-agent` is only a `devDependency`.

That's not what I call a "backward compatible" update.

**Comment by tstackhouse** [comment]
My workaround was to pin to 1.9.0, which was the version we were using in my environment as it was.

**Comment by sathyarajagopal** [comment]
+1 Until 1.10.1 is published, we are using 1.9.0 for now

**Comment by jandppw** [comment]
the latest tag MUST be removed from 1.10.0 in npm immediately

**Comment by cullylarson** [comment]
I suspect the issue is that `https-proxy-agent` is set as a devDependency.

**Comment by mendhak** [comment]
Several of our services failed today due to this; we're using this library in the AWS API Gateway Authorizer Lambdas, and same as tstackhouse, builds don't fail but the Lambdas crash after deployment.

**Comment by davidpatrick** [comment]
Hey all sorry about the release yesterday.  As soon as I logged on this morning I saw the misplaced dependency and PRs to fix it.  The fix is in 1.10.1 and the release has been published to npmjs.

**Comment by FabianEbert** [comment]
Thank you for releasing a fix so quickly 👍
